File: Solutions/0868/0868.ts
function binaryGap(n: number): number {
    let ans:number = 0;

    for (let d = -32; n; n >>= 1, ++d) {
        if (n % 2 === 1) {
            ans = Math.max(ans, d);
            d = 0;
        }
    }

    return ans;
};

File: Solutions/0868/0868.go
func binaryGap(n int) int {
	var ans int

	for d := -32; n > 0; n >>= 1 {
		if n%2 == 1 {
			ans = max(ans, d)
			d = 0
		}
		d++
	}

	return ans
}